v2/client/client.go

```diff
@@ -31,12 +31,37 @@ const DefaultUserAgent = "voucher-client/2"
 
 // NewClient creates a new Client set to connect to the passed
 // hostname.
-func NewClient(voucherURL string) (*Client, error) {
-	return newClient(voucherURL, &http.Client{})
+func NewClient(voucherURL string, opts ...Option) (*Client, error) {
+	return newClient(voucherURL, &http.Client{}, opts...)
+}
+
+type Option func(*Client)
+
+// WithHTTPClient sets the http.Client to use for the client.
+func WithHTTPClient(httpClient *http.Client) Option {
+	return func(c *Client) {
+		c.httpClient = httpClient
+	}
+}
+
+// WithUserAgent sets the User-Agent header for the client.
+func WithUserAgent(userAgent string) Option {
+	return func(c *Client) {
+		c.userAgent = userAgent
+	}
+}
+
+// WithBasicAuth sets the username and password to use for the client.
+func WithBasicAuth(username, password string) Option {
+	return func(c *Client) {
+		c.username = username
+		c.password = password
+	}
 }
 
 // NewAuthClient creates a new auth Client set to connect to the passed
 // hostname using tokens.
+// Deprecated: use the WithHTTPClient option instead
```

Comment:
Are we expecting users to set up the auth client with `idtoken` and then call `WithHTTPClient`? I think this comment is unclear that `WithHTTPClient` will not handle auth.
Maybe we should keep this function to be used with `WithHTTPClient` for clarity, unless it is only us who would need to use this kind of auth (but then the comment should indicate that).
We could also have `WithAuthClient` option like this:
```go
func WithAuthClient(httpClient *http.Client) Option {
        authClient, err := idtoken.NewClient(context.Background(), voucherURL)
        authClient.url = httpClient.url
        authClient.userAgent = httpClient.userAgent
	return func(c *Client) {
		c.httpClient = authClient
	}
}
```
But this could get inconsistent if we change newClient.

Reply:
> Are we expecting users to set up the auth client with idtoken and then call WithHTTPClient

Yep! Wrapping `http.Transport` to customize request payloads (e.g. to inject authentication credentials) is common, but I've added an explicit comment.

> We could also have WithAuthClient option like this

I like the idea of not leaving _everything_ about the client's initialization to the caller: it seemed OK in this PR's scope (since the CLI is updated), but it doesn't work well when voucher is included as an SDK.

This became `WithIDTokenAuth()`, which is darned close to the `WithAuthClient` you drafted:
* I made `Option` accept a `context.Context` argument, so callers can specify timeouts/interrupts
* I made `Option` return an error, so `idtoken.NewClient()` errors will be returned.